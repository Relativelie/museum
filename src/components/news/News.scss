@import "../../variables";

.news {
    display: grid;
    grid-column-gap: 20px;
    grid-row-gap: 50px;
    grid-template-columns: repeat(auto-fill, 400px);
    justify-content: space-between;
    margin: 0 100px;

    & .news_title,
    .news_description {
        font-family: $fontFamily;
        // color: $fontColor;
    }

    & .news_title {
        margin: 0 0 10px 0;
        font-size: 16px;
    }

    & .news_description {
        margin: 0;
        font-size: 14px;
    }
}