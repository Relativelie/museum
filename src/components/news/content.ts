import imageOne from '../../assets/images/news/newsOne.png';
import imageTwo from '../../assets/images/news/newsTwo.png';
import imageThree from '../../assets/images/news/newsThree.png';
import imageFour from '../../assets/images/news/newsFour.png';
import imageFive from '../../assets/images/news/newsFive.png';
import imageSix from '../../assets/images/news/newsSix.png';

export const content = [
    {
        id: 4001,
        title: 'Заголовок',
        description: 'ОписаниОписание Описание  Описание Описание Описание  Описаниее',
        image: imageOne,
    },
    {
        id: 4002,
        title: 'Заголовок',
        description: 'ОписаниОписание Описание  Описание Описание Описание  Описаниее',
        image: imageTwo,
    },
    {
        id: 4003,
        title: 'Заголовок',
        description: 'ОписаниОписание Описание  Описание Описание Описание  Описаниее',
        image: imageThree,
    },
    {
        id: 4004,
        title: 'Заголовок',
        description: 'ОписаниОписание Описание  Описание Описание Описание  Описаниее',
        image: imageFour,
    },
    {
        id: 4005,
        title: 'Заголовок',
        description: 'ОписаниОписание Описание  Описание Описание Описание  Описаниее',
        image: imageFive,
    },
    {
        id: 4006,
        title: 'Заголовок',
        description: 'ОписаниОписание Описание  Описание Описание Описание  Описаниее',
        image: imageSix,
    },
];
