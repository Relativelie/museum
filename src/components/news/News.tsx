import { FC } from 'react';
import { content } from './content';

import './News.scss';

export const News: FC = () => {
    return (
        <div className="news">
            <h3 className="newsTitle">Новости</h3>
            <div className="newsContent">
                {content.map(((elem) => {
                    const { id, title, description, image } = elem;
                    return (
                        <div className="newsContent_item" key={`newsContent-${id}`}>
                            <img src={image} alt="news" />
                            <h4 className="newsContent_title">{title}</h4>
                            <p className="newsContent_description">{description}</p>
                        </div>
                    );
                }))}
            </div>
        </div>
    );
};
