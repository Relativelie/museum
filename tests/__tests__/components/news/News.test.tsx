import '@testing-library/jest-dom/extend-expect';
import '@testing-library/jest-dom';
import * as ShallowRenderer from 'react-test-renderer/shallow';

import { News } from '../../../../src/components/news/News';

describe('Feedback - Feedback component', () => {
    test('snapshot - Feedback component', () => {
        const renderer = ShallowRenderer.createRenderer();
        renderer.render(<News />);
        const tree = renderer.getRenderOutput();
        expect(tree).toMatchSnapshot();
    });
});
