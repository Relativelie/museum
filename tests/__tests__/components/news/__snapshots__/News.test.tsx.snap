// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback - Feedback component snapshot - Feedback component 1`] = `
<div
  className="news"
>
  <h3
    className="newsTitle"
  >
    Новости
  </h3>
  <div
    className="newsContent"
  >
    <div
      className="newsContent_item"
    >
      <img
        alt="news"
        src="test-file-stub"
      />
      <h4
        className="newsContent_title"
      >
        Заголовок
      </h4>
      <p
        className="newsContent_description"
      >
        ОписаниОписание Описание  Описание Описание Описание  Описаниее
      </p>
    </div>
    <div
      className="newsContent_item"
    >
      <img
        alt="news"
        src="test-file-stub"
      />
      <h4
        className="newsContent_title"
      >
        Заголовок
      </h4>
      <p
        className="newsContent_description"
      >
        ОписаниОписание Описание  Описание Описание Описание  Описаниее
      </p>
    </div>
    <div
      className="newsContent_item"
    >
      <img
        alt="news"
        src="test-file-stub"
      />
      <h4
        className="newsContent_title"
      >
        Заголовок
      </h4>
      <p
        className="newsContent_description"
      >
        ОписаниОписание Описание  Описание Описание Описание  Описаниее
      </p>
    </div>
    <div
      className="newsContent_item"
    >
      <img
        alt="news"
        src="test-file-stub"
      />
      <h4
        className="newsContent_title"
      >
        Заголовок
      </h4>
      <p
        className="newsContent_description"
      >
        ОписаниОписание Описание  Описание Описание Описание  Описаниее
      </p>
    </div>
    <div
      className="newsContent_item"
    >
      <img
        alt="news"
        src="test-file-stub"
      />
      <h4
        className="newsContent_title"
      >
        Заголовок
      </h4>
      <p
        className="newsContent_description"
      >
        ОписаниОписание Описание  Описание Описание Описание  Описаниее
      </p>
    </div>
    <div
      className="newsContent_item"
    >
      <img
        alt="news"
        src="test-file-stub"
      />
      <h4
        className="newsContent_title"
      >
        Заголовок
      </h4>
      <p
        className="newsContent_description"
      >
        ОписаниОписание Описание  Описание Описание Описание  Описаниее
      </p>
    </div>
  </div>
</div>
`;
